easybuild/framework/easyconfig/tweak.py

```diff
@@ -833,6 +950,71 @@ def map_easyconfig_to_target_tc_hierarchy(ec_spec, toolchain_mapping, targetdir=
+
+                    # determine main install version based on toolchain
+                    if toolchain['name'] != DUMMY_TOOLCHAIN_NAME:
+                        toolchain_suffix = "-%s-%s" % (toolchain['name'], toolchain['version'])
+                    else:
+                        toolchain_suffix = ''
+                    full_versionsuffix = ''.join([x for x in [toolchain_suffix, versionsuffix, EB_FORMAT_EXTENSION]
+                                                  if x])
+
+                    # prepend/append version prefix/suffix
+                    depver = ''.join([x for x in ['^', versionprefix, candidate_ver, full_versionsuffix] if x])
+                    cand_paths = search_easyconfigs(depver, return_robot_list=True)
+                    # Add them to the possibilities
+                    for path in cand_paths:
+                        # Get the version from the path
+                        filename = os.path.basename(path)
+                        # Find the version sandwiched between our known values
+                        try:
+                            regex = '^%s(.+?)%s' % (versionprefix, full_versionsuffix)
+                            version = re.search(regex, filename).group(1)
+                        except AttributeError:
+                            raise EasyBuildError("Somethings wrong, could not extract version from %s using %s",
+                                                 filename, regex)
+                        potential_version_matches.append({'version': version, 'path': path, 'toolchain': toolchain})
+                _log.debug("Found possible dependency upgrades: %s\n", potential_version_matches)
+
+                # Compare this version to the original versions and replace if appropriate (upgrades only)
+                highest_version = dep['version']
+                for candidate in potential_version_matches:
+                    if LooseVersion(candidate['version']) >= LooseVersion(highest_version):
+                        highest_version = candidate['version']
+                if highest_version != dep['version']:
+                    _log.info("Increasing version to %s for dependency %s.", highest_version, orig_dep['name'])
+                    _log.info("Depending on your configuration, this will be resolved with one of the following "
+                              "easyconfigs: %s", '\n'.join(cand['path'] for cand in potential_version_matches
+                                                           if cand['version'] == highest_version))
+                    orig_dep['version'] = highest_version
+                    if orig_dep['versionsuffix'] in versonsuffix_mapping:
+                        orig_dep['versionsuffix'] = versonsuffix_mapping[orig_dep['versionsuffix']]
+                    orig_dep['short_mod_name'] = ActiveMNS().det_short_module_name(dep)
+                    orig_dep['full_mod_name'] = ActiveMNS().det_full_module_name(dep)
```

Comment:
@ocaisa We're also doing this above, can we avoid this code duplication?

Something like:

```python
if 'binutils' in ...:
    ...
    dep_changed = True
elif update_dep_versions:
    ...
    dep_changed = True

if dep_changed:
    orig_dep['short_mod_name'] = ActiveMNS().det_short_module_name(dep)
    orig_dep['full_mod_name'] = ActiveMNS().det_full_module_name(dep)
    _log.info("Updated module names for dep %s v%s: %s (short), %s (long)",
              orig_dep['name'], orig_dep['version'], orig_dep['short_mod_name'], orig_dep['full_mod_name'])